pkg/manager/internal.go

```diff
@@ -135,22 +150,44 @@ func (cm *controllerManager) Add(r Runnable) error {
 		return err
 	}
 
-	var shouldStart bool
+	if cm.leaderElectionRunnables == nil {
+		cm.leaderElectionRunnables = make(map[string][]Runnable)
+	}
 
 	// Add the runnable to the leader election or the non-leaderelection list
-	if leRunnable, ok := r.(LeaderElectionRunnable); ok && !leRunnable.NeedLeaderElection() {
-		shouldStart = cm.started
-		cm.nonLeaderElectionRunnables = append(cm.nonLeaderElectionRunnables, r)
+	if leRunnable, ok := r.(LeaderElectionRunnable); ok && leRunnable.NeedLeaderElection() {
```

Comment:
this was actually intentional (we should've had a comment here) -- we default to "use leader election" intentionally to avoid breaking existing code.